Develop1

frozmoroz2017 opened the pull request.

frozmoroz2017 committed:
1

frozmoroz2017 committed:
1

frozmoroz2017 committed:
dd aoo test coverage report

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,224 @@
+     private Bun bun;
+
+     @Mock
+     private Ingredient ingredient1;
```
⛔️Нужно исправить. При нейминге не рекомендуется использовать числа (Field2), их еще называют magicNumbers. Очень тяжело поддерживать код с magicNumbers.

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,224 @@
+
+     @Test
+     public void testSetBuns() {
+         Burger burger = new Burger();
```
⚠️Можно улучшить. Инициализацию объекта можно вынести в метод с аннотацией before

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,224 @@
+         Burger burger = new Burger();
+         burger.addIngredient(ingredient1);
+
+         assertEquals(1, burger.ingredients.size());
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,224 @@
+     @RunWith(Parameterized.class)
+     public static class GetPriceParameterizedTest {
+
+         @Parameterized.Parameters
```
⚠️Можно улучшить. В параметризованных тестах для аннотации Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,224 @@
+
+         String receipt = burger.getReceipt();
+
+         assertTrue(receipt.contains("Price: 20,000000"));
```
⛔️Нужно исправить. Для юнит-тестов применим подход: один тест, значит одна проверка. В этом тесте две проверки (Mockito.verify, assertEquals), а должна быть одна. Исправь, пожалуйста, этот момент во всем коде.

frozmoroz2017 committed:
исправлено:
- поменял magic numbers в названиях переменных;
- азделил тесты с несколькими проверками на отдельные тесты
- реализовл принцип 'один тест - одна проверка'
